### PR TITLE
fix(helm-port): add input name length/regex rules

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-1180

https://www.loom.com/share/dd4457fb5f9f43a1bf0c21cbcbe6cc84

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [ ] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> The backlog never ends, be at peace with it

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] test-remi2 - fix(helm-port): add input name length and regex rules - 2024-06-20T16:38:02Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/2de3e145-d71b-4dd8-a78d-0d834316ee5d/services/general)

 Another comment will be posted when deployments are finished

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment failed to be deployed !
Click on the link below to check the deployment logs.
:point_right: [\[PR\] test-remi2 - fix(helm-port): add input name length and regex rules - 2024-06-20T16:38:02Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/2de3e145-d71b-4dd8-a78d-0d834316ee5d/logs)

File: libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx [hunk 1]
```diff
@@ -70,12 +82,16 @@ export function NetworkingPortSettingModal({ port, onClose, onSubmit }: Networki
...
               required: 'Please enter service name.',
+              ...namePatternRules,
             }}
             render={({ field, fieldState: { error } }) => (
               <InputText
                 name={field.name}
                 onChange={(e: FormEvent<HTMLInputElement>) => {
-                  setValue('name', `${e.currentTarget.value}-p${watchInternalPort}`)
+                  const name = `${e.currentTarget.value}-p${watchInternalPort}`
+                  if (name.length < 38) {
```
Comment (ctjhoa) [inline libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx:92]: Can't you use `maxlength` props on the input directly?

Comment (RemiBonnet) [reply libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx]: I don't find anything in the library and documentation, but I can do something like this: https://github.com/Qovery/console/pull/1444/commits/69debcbf4d31480d0a25dd42d0881eddca36d5d4

Comment (ctjhoa) [reply libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx]: I was thinking about the html props https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/maxlength
no library involve

Comment (RemiBonnet) [reply libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx]: I tried to add `maxLength` on the input but the `setValue` don't take into account this rule. It works only if I edit the input directly not from an `setValue`

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**